src/client.ts

```diff
@@ -107,171 +108,182 @@ export class TwilioServerlessApiClient extends events.EventEmitter {
-            const environment = await getEnvironment(
-              environmentSid,
-              serviceSid,
-              this.client
-            );
-            currentBuildSidForEnv = environment.build_sid;
           }
 
-          if (type === 'functions' || type === 'assets') {
-            if (!currentBuild) {
-              currentBuild = await getBuild(
-                currentBuildSidForEnv,
+          if (type === 'builds') {
+            result.builds = await listBuilds(serviceSid, this.client);
+          }
+
+          if (typeof environmentSid === 'string') {
+            if (!isEnvironmentSid(environmentSid)) {
+              const environment = await getEnvironmentFromSuffix(
+                environmentSid,
                 serviceSid,
                 this.client
               );
+              environmentSid = environment.sid;
+              currentBuildSidForEnv = environment.build_sid;
+            } else if (!currentBuildSidForEnv) {
+              const environment = await getEnvironment(
+                environmentSid,
+                serviceSid,
+                this.client
+              );
+              currentBuildSidForEnv = environment.build_sid;
             }
 
-            if (type === 'functions') {
-              result.functions = {
-                environmentSid,
-                entries: currentBuild.function_versions,
-              };
-            } else if (type === 'assets') {
-              result.assets = {
-                environmentSid,
-                entries: currentBuild.asset_versions,
-              };
+            if (type === 'functions' || type === 'assets') {
+              if (!currentBuild) {
+                currentBuild = await getBuild(
+                  currentBuildSidForEnv,
+                  serviceSid,
+                  this.client
+                );
+              }
+
+              if (type === 'functions') {
+                result.functions = {
+                  environmentSid,
+                  entries: currentBuild.function_versions,
+                };
+              } else if (type === 'assets') {
+                result.assets = {
+                  environmentSid,
+                  entries: currentBuild.asset_versions,
+                };
+              }
             }
-          }
 
-          if (type === 'variables') {
-            result.variables = {
-              entries: await listVariablesForEnvironment(
+            if (type === 'variables') {
+              result.variables = {
+                entries: await listVariablesForEnvironment(
+                  environmentSid,
+                  serviceSid,
+                  this.client
+                ),
                 environmentSid,
-                serviceSid,
-                this.client
-              ),
-              environmentSid,
-            };
+              };
+            }
           }
+        } catch (err) {
+          log(new ClientApiError(err));
```

Comment:
I know this isn't a full change as the previous version just logged the error too, but this swallows the error and allows the code to carry on and all other functions in this file eventually throw their errors. Should this be logged? Or should it throw and be caught and re-raised by line 213 (which is new to this function)?

Reply:
I think for now we'll keep the error swallowing logic because it's being used in a for loop and it ensures that even if one type you are trying to list fails it can still return the others.